src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -92,114 +147,1104 @@ static DamageTable D_8098C33C = {
+            func_80989140(&this->skelAnime, D_8098C35C, 2);
+        } else if (D_8098C2A8[this->unk_286].unk_04 & 0x11) {
+            func_80989140(&this->skelAnime, D_8098C35C, 1);
+        } else {
+            func_80989140(&this->skelAnime, D_8098C35C, 2);
+        }
+        this->actionFunc = func_8098A468;
+    }
+}
+
+void func_80989BF8(EnDg* this) {
+    if (this->unk_286 < 14) {
+        if (this->unk_286 % 2) {
+            D_8098C2A8[this->unk_286].unk_04 =
+                0x3538 + ((gSaveContext.weekEventReg[42 + (this->unk_286 / 2)] & 0xF0) >> 4);
+        } else {
+            D_8098C2A8[this->unk_286].unk_04 = 0x3538 + (gSaveContext.weekEventReg[42 + (this->unk_286 / 2)] & 0xF);
+        }
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+
+    if ((D_8098C2A8[this->unk_286].unk_04 >= 0x3547) || (D_8098C2A8[this->unk_286].unk_04 < 0x3538)) {
+        D_8098C2A8[this->unk_286].unk_04 = 0x353E;
+    }
+
+    if (D_8098C2A8[this->unk_286].unk_04 == 0x353D) {
+        D_8098C2A8[this->unk_286].unk_04 = 0x3538;
+    }
+}
+
+void func_80989D38(EnDg* this, GlobalContext* globalCtx) {
+    if (this->unk_286 == 21) {
+        if ((gSaveContext.day % 5) == 1) {
+            func_801518B0(globalCtx, 0x91C, NULL);
+        } else {
+            func_801518B0(globalCtx, 0x91E, NULL);
+        }
+    } else if ((this->unk_286 >= 0) && (this->unk_286 < 14)) {
+        func_801518B0(globalCtx, D_8098C2A8[this->unk_286].unk_04, NULL);
+    } else if (this->unk_286 == 20) {
+        func_801518B0(globalCtx, 0x353D, NULL);
+    } else {
+        func_801518B0(globalCtx, 0x627, NULL);
+    }
+}
+
+void func_80989E18(EnDg* this, GlobalContext* globalCtx) {
+    D_8098C2A8_s* temp;
+
+    if ((D_8098C2A0 != 0) && !(this->unk_280 & 1)) {
+        this->actor.flags |= 0x8000000;
+        this->unk_280 |= 1;
+    } else if ((D_8098C2A0 == 0) && (this->unk_280 & 1)) {
+        this->actor.flags &= ~0x8000000;
+        this->unk_280 &= ~1;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_SMALL_DOG_BARK);
+        this->unk_290 = 1;
+        D_8098C2FC = D_8098C2A8[this->unk_286];
+        if (D_8098C2A0 == 0) {
+            this->actor.flags |= 0x8000000;
+            D_8098C2A0 = 1;
+            this->unk_280 |= 1;
+        }
+
+        func_80989140(&this->skelAnime, D_8098C35C, 5);
+        this->actor.flags &= ~1;
+        this->actor.speedXZ = 0.0f;
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_MASK_OF_TRUTH) {
+            this->actor.flags |= 0x10000;
+            func_800B8614(&this->actor, globalCtx, 100.0f);
+            this->actionFunc = func_8098BBEC;
+        } else {
+            this->actionFunc = func_8098BA64;
+        }
+    } else {
+        func_800B8BB0(&this->actor, globalCtx);
+    }
+}
+
+s32 func_80989FC8(GlobalContext* globalCtx) {
+    EnDg* enemy = (EnDg*)globalCtx->actorCtx.actorList[ACTORCAT_ENEMY].first;
+    f32 maxDist = 9999.0f;
```

Comment:
This should be "minDist", because this function seems to be looking for the minimal distance (see the `if (dist < maxDist) {` comparison)